autoload/ale/events.vim

```diff
@@ -138,6 +138,32 @@ function! ale#events#Init() abort
                 " not be changed here.
                 autocmd InsertLeave * if exists('*ale#engine#Cleanup') | call ale#cursor#EchoCursorWarning() | endif
             endif
+
+            function! AleShowCursorDetailCmd() abort
+                if exists('*ale#engine#Cleanup')
+                    call ale#preview#CloseIfTypeMatches('ale-preview')
+                    " Don't jump to the preview window during cursor movement.
+                    call ale#cursor#ShowCursorDetail({'stay_here': 1})
+                endif
+            endfunction
+
+            if g:ale_show_cursor_detail
+                if len(g:ale_cursor_detail_fileextensions)
```

Comment:
Instead of adding an option to specify which file extensions to enable the option for, use `ale#Var` to get either a `b:` or `g:` variable for the option, and people can turn the option on in `ftplugin` files. That's how most of the other ALE options work.

Remove the additional `autocmd` command here, and check for if we should automatically open the preview window in the `ale#cursor#EchoCursorWarning` instead. The `InsertEnter` event will probably still be useful, but you should also check if we should close the preview when the cursor moves off the line in `ale#cursor#EchoCursorWarning`.

Reply:
What if the person doesn't have g:ale_echo_cursor set? It wouldn't enter the if there:

```
 if g:ale_echo_cursor                                                                                                                         
    autocmd CursorMoved,CursorHold * if exists('*ale#engine#Cleanup') | call ale#cursor#EchoCursorWarningWithDelay() | endif
```

Which is why I have it separated under a different block with an extra AutoCmd instead. Do you want to rename EchoCursorWarning to some other function that does both?

By the way, feel free to push commits to this branch and organize the code however you see fit. I'm sadly not able to follow what are your intentions with those tweaks. If you want to close this PR and do this by yourself later, that's fine too.

Reply:
If `g:ale_echo_cursor` isn't enabled, you don't get detailed messages either.